vtkLookingGlassInterface.cxx

```diff
@@ -627,6 +628,132 @@ void vtkLookingGlassInterface::GetTilePosition(int tile, int pos[2])
+  std::vector<vtkCamera*> Cameras;
+  for (renCol->InitTraversal(rsit); aren = renCol->GetNextRenderer(rsit);)
+  {
+    // Ugly piece of code - we need to know if the camera already
+    // exists or not. If it does not yet exist, we must reset the
+    // camera here - otherwise it will never be done (missing its
+    // oppportunity to be reset in the Render method of the
+    // vtkRenderer because it will already exist by that point...)
+    if (!aren->IsActiveCameraCreated())
+    {
+      aren->ResetCamera();
+    }
+    auto oldCam = aren->GetActiveCamera();
+    oldCam->SetLeftEye(1);
+    oldCam->Register(rw);
+    Cameras.push_back(oldCam);
+    vtkNew<vtkCamera> newCam;
+    aren->SetActiveCamera(newCam);
+  }
+
+  // Make sure we can use offscreen buffers
+  auto prevUseOffScreenBuffers = rw->GetUseOffScreenBuffers();
+  rw->UseOffScreenBuffersOn();
+
+  // loop over all the tiles and render then and blit them to the quilt
+  for (int tile = 0; tile < tcount; ++tile)
+  {
+    renderFramebuffer->Bind(GL_DRAW_FRAMEBUFFER);
+    ostate->vtkglViewport(0, 0, renderSize[0], renderSize[1]);
+    ostate->vtkglScissor(0, 0, renderSize[0], renderSize[1]);
+
+    int count = 0;
+    for (renCol->InitTraversal(rsit); aren = renCol->GetNextRenderer(rsit); ++count)
+    {
+      // adjust camera
+      vtkCamera* cam = aren->GetActiveCamera();
+      cam->DeepCopy(Cameras[count]);
+      this->AdjustCamera(cam, tile);
+
+      // limit the clipping range to limit parallax
+      if (this->GetUseClippingLimits())
+      {
+        double* cRange = cam->GetClippingRange();
+        double cameraDistance = cam->GetDistance();
+
+        double nearClippingLimit = this->GetNearClippingLimit();
+        double farClippingLimit = this->GetFarClippingLimit();
+
+        double newRange[2];
+        newRange[0] = cRange[0];
+        newRange[1] = cRange[1];
+        if (cRange[0] < cameraDistance * nearClippingLimit)
+        {
+          newRange[0] = cameraDistance * nearClippingLimit;
+        }
+        if (cRange[1] > cameraDistance * farClippingLimit)
+        {
+          newRange[1] = cameraDistance * farClippingLimit;
+        }
+        cam->SetClippingRange(newRange);
+      }
+    }
+
+    if (renderFunc)
+    {
+      (*renderFunc)();
+    }
+    else
+    {
+      renCol->Render();
+    }
+
+    quiltFramebuffer->Bind(GL_DRAW_FRAMEBUFFER);
+
+    int destPos[2];
+    this->GetTilePosition(tile, destPos);
+
+    // blit to quilt
+    ostate->vtkglViewport(destPos[0], destPos[1], renderSize[0], renderSize[1]);
+    ostate->vtkglScissor(destPos[0], destPos[1], renderSize[0], renderSize[1]);
+    glBlitFramebuffer(0, 0, renderSize[0], renderSize[1], destPos[0], destPos[1],
+      destPos[0] + renderSize[0], destPos[1] + renderSize[1], GL_COLOR_BUFFER_BIT, GL_LINEAR);
+  }
+  ostate->PopFramebufferBindings();
+
+  if (this->IsRecording)
+  {
+    // Write out a movie frame if we are recording
+    this->WriteQuiltMovieFrame();
+  }
```

Comment:
This was originally just in ParaView, but will be useful for the other two cases as well if we want to add the ability to record movies to them.